hydrolib/core/io/gui/models.py

```diff
@@ -0,0 +1,125 @@
+        branchtype: Optional[str] = Field(
+            "Channel = 0, SewerConnection = 1, Pipe = 2.", alias="branchType"
+        )
+        islengthcustom: Optional[str] = Field(
+            "branch length specified by user.", alias="isLengthCustom"
+        )
+        sourcecompartmentname: Optional[str] = Field(
+            "Source compartment name this sewer connection is beginning.",
+            alias="sourceCompartmentName",
+        )
+        targetcompartmentname: Optional[str] = Field(
+            "Source compartment name this sewer connection is beginning.",
+            alias="targetCompartmentName",
+        )
+        material: Optional[str] = Field(
+            "0 = Unknown, 1 = Concrete, 2 = CastIron, 3 = StoneWare, 4 = Hdpe, "
+            "5 = Masonry, 6 = SheetMetal, 7 = Polyester, 8 = Polyvinylchlorid, 9 = Steel"
+        )
+
+    comments: Comments = Comments()
+
+    _header: Literal["Branch"] = "Branch"
+
+    name: str = Field("name", max_length=255, alias="name")
+    branchtype: int = Field(0, alias="branchType")
+    islengthcustom: Optional[bool] = Field(True, alias="isLengthCustom")
+    sourcecompartmentname: Optional[str] = Field(None, alias="sourceCompartmentName")
+    targetcompartmentname: Optional[str] = Field(None, alias="targetCompartmentName")
+    material: Optional[int] = Field(None, alias="material")
+
+    def _get_identifier(self, data: dict) -> Optional[str]:
+        return data.get("name")
+
+    @validator("branchtype")
+    def _validate_branchtype(cls, branchtype: int):
+        if branchtype not in [0, 1, 2]:
+            raise ValueError(f"branchtype is not allowed {branchtype}")
+        return branchtype
+
+    @validator("material")
+    def _validate_material(cls, material: int):
+        if material not in [range(10)]:
+            raise ValueError("material is not allowed")
+        return material
+
+
+class BranchModel(INIModel):
+    """
+    The overall branch model that contains the contents of one branches.gui file.
+
+    This model is not referenced under a [FMModel][hydrolib.core.io.mdu.models.FMModel].
+
+    Attributes:
+        branch(List[Branch]): List of `[Branch]` blocks for all branches.
+    """
+
+    general: BranchGeneral = BranchGeneral()
+    branch: List[Branch] = []
+
+    _make_list = make_list_validator("branch")
+
+    @classmethod
+    def _ext(cls) -> str:
+        return ".gui"
+
+    @classmethod
+    def _filename(cls) -> str:
+        return "branches"
+
+    @validator("branch", each_item=True)
+    def _validate(cls, branch: Branch):
+        """Validates for each pipe whether the compartment info is filled in"""
+
+        if branch.branchtype == 2 and (
+            branch.sourcecompartmentname is None
+            and branch.targetcompartmentname is None
+        ):
+            raise ValueError(
+                f"Either source or target compartment name should be provided when branchtype is 2 for branch id {branch.name}"
```

Comment:
```suggestion
                f"Either sourceCompartmentName or targetCompartmentName should be provided when branchType is 2 for branch with name {branch.name}"
```

Hydrolib-core follows the convention to use the exact names from the files. 
Also a question, why should the source and target compartment name not be be provided both?